### PR TITLE
rewrite of rancher docker image and chart promotion script

### LINKED ISSUE #247: Automate tagging the latest GA release of rancher/rancher to also be the latest tag for rancher/rancher
This one might be a bit more complex as we publish a lot of different release lines, but we manually have to update the latest docker image for rancher/rancher to the latest version of Rancher. 

This doesn't have to be in drone step but could be based on a GH action or some other tool, but not having to manually run a script.

### PR DESCRIPTION
**Pull request by briandowns**
Resolves #247 

Adds dry-run capabilities.

### CODE REVIEW

File: bin/rancher_image_chart_promote.sh [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+case $2 in 
+    "stable")
+    ;;
+    "latest")
+    ;;
+    "donotuse")
+    ;;
+    "*")
+        echo "error: tag needs to be stable, latest, or donotuse for testing. given: $2"
```
Comment (luisrojas2) [inline bin/rancher_image_chart_promote.sh:50]: is this a typo?

Comment (briandowns) [reply bin/rancher_image_chart_promote.sh:50]: Not that I'm aware of.

File: bin/rancher_image_chart_promote.sh [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+until [ ${PAGE} -gt 100 ]; do
+    echo "finding build number for tag: ${SOURCE_TAG}"
+
+    BUILD_NUMBER=$(drone build ls rancher/rancher --page ${PAGE} --event tag --format "{{.Number}},{{.Ref}}"| grep "${SOURCE_TAG}"$ |cut -d',' -f1|head -1)
+    if [ ! -n "${BUILD_NUMBER}" ]; then
+        echo "error: no build found for tag: ${SOURCE_TAG}"
+        exit 1
+    fi
+    
+    PAGE=$((PAGE+1))
```
Comment (brooksn) [inline bin/rancher_image_chart_promote.sh:67]: It looks like there's no break condition if a build is found.

File: bin/rancher_image_chart_promote.sh [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    *)
+        usage
+        exit 1
+    ;;
+    esac
+done
+
+has_drone
+
+if [ $# -ne 2 ]; then
```
Comment (brooksn) [inline bin/rancher_image_chart_promote.sh:35]: I reach this condition if I run `./rancher_image_chart_promote.sh -d v2.7.1 latest`. What's the right command for a dry run?

File: bin/rancher_image_chart_promote.sh [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+
+    sleep 1
+done
+
+echo "Found build number ${BUILD_NUMBER} for tag ${SOURCE_TAG}"
+if [ "${DRY_RUN}" = true ]; then
+    CMD="drone build promote rancher/rancher ${BUILD_NUMBER} promote-docker-image --param=SOURCE_TAG=${SOURCE_TAG} --param=DESTINATION_TAG=${DESTINATION_TAG}"
+    echo "${CMD}"
+else
+    drone build promote rancher/rancher "${BUILD_NUMBER} promote-docker-image --param=SOURCE_TAG=${SOURCE_TAG} --param=DESTINATION_TAG=${DESTINATION_TAG}"
```
Comment (brooksn) [inline bin/rancher_image_chart_promote.sh:77]: Could there need to be a sleep here, so the resulting build is listed in the next `drone build ls` command?

File: bin/rancher_image_chart_promote.sh (hunk 3 above)
Comment (briandowns) [reply bin/rancher_image_chart_promote.sh]: I've removed dry run for now as it's not really behaving as I'd expect to. I'll add it back when this code gets used. The active code is in the rancher repo.

**Review by brooksn — APPROVED** [review]

**Pull request merged**